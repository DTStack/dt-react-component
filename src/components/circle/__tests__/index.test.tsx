
import React from 'react';
import Circle from '../index';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

describe('test Circle suite', () => {
    test('should support circle success render', () => {
        const wrapper = render(<Circle>完成</Circle>)
        expect(wrapper).toMatchSnapshot();
    })
    test('should support circle render the default className in Circle', () => {
        const { container } = render(<Circle>完成</Circle>)
        expect(container.firstChild).toHaveClass(...['dtc-circle', 'dtc-circle-border']);
    })
    test('should render circle render correct type', () => {
        const { container } = render(<Circle type="run">完成</Circle>)
        expect(container.firstChild.firstChild).toHaveClass('dtc-circle-run');
    })
    test('should render circle render correct color', () => {
        const { container } = render(<Circle color="#2177b8">完成</Circle>)
        const textWapper = container.firstChild.firstChild
        expect(textWapper).toHaveStyle({ background: '#2177b8' });
    })
    test('should render circle render correct text', () => {
        const { container } = render(<Circle>自定义文案</Circle>)
        const textWapper = container.querySelector('.dtc-circle-text')
        expect(textWapper.innerHTML).toEqual('自定义文案');
    })
})
