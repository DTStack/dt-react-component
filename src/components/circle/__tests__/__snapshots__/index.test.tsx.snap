// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Circle suite should support circle success render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dtc-circle dtc-circle-border"
      >
        <div
          class="dtc-circle-default dtc-circle-finished"
        />
        <span
          class="dtc-circle-text"
        >
          完成
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dtc-circle dtc-circle-border"
    >
      <div
        class="dtc-circle-default dtc-circle-finished"
      />
      <span
        class="dtc-circle-text"
      >
        完成
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
